components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java

```diff
@@ -104,18 +104,22 @@ public GatewayStartupListener() {
 
     @Override
     public void completingServerStartup() {
-
-        try {
-            CertificateMgtUtils.backupOriginalTrustStore();
-            CertificateMgtUtils.startListenerCertificateReLoader();
-        } catch (CertificateManagementException e) {
-            log.error("Error while Backup Truststore", e);
+        String migrationEnabled = System.getProperty(APIConstants.MIGRATE);
+        if (migrationEnabled == null) {
+            try {
+                CertificateMgtUtils.backupOriginalTrustStore();
+                CertificateMgtUtils.startListenerCertificateReLoader();
+            } catch (CertificateManagementException e) {
+                log.error("Error while Backup Truststore", e);
+            }
+            log.debug("Registering ServerStartupListener for SubscriptionStore for the tenant domain : " + MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            SubscriptionDataHolder.getInstance().registerTenantSubscriptionStore(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            log.debug("Registered ServerStartupListener for SubscriptionStore for the tenant domain : " + MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            SubscriptionDataHolder.getInstance().initializeSubscriptionStore(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            cleanDeployment(CarbonUtils.getCarbonRepository());
+        } else {
+            log.info("Stopped at gateway startup listener completing!");
```

Comment:
Shall we make log more descriptive by adding that "running on migration enabled mode"